search/api.py

```diff
@@ -316,6 +317,42 @@ def find_related_documents(*, user, post_id):
     return search.execute().to_dict()
 
 
+def find_similar_resources(*, user, value_doc):
+    """
+    Execute a "more like this" query to find learning resources that are similar to the one provided.
+
+     Args:
+        user (User): The user executing the search
+        value_doc (dict):
+            a document representing the data fields we want to search with
+
+    Returns:
+        dict: The Elasticsearch response dict
+    """
+    index = get_default_alias_name(ALIAS_ALL_INDICES)
+    search = Search(index=index, using=get_conn())
+    search = _apply_general_query_filters(search, user)
+    search = search.filter(Q("terms", object_type=LEARNING_RESOURCE_TYPES))
+    search = search.query(
+        MoreLikeThis(
+            like={"doc": value_doc, "fields": list(value_doc.keys())},
+            fields=SIMILAR_RESOURCE_RELEVANT_FIELDS,
+            min_term_freq=settings.OPEN_RESOURCES_MIN_TERM_FREQ,
+            min_doc_freq=settings.OPEN_RESOURCES_MIN_DOC_FREQ,
+        )
+    )
+    response = search.execute()
+
+    return [
+        hit.to_dict()
+        for hit in response.hits
+        if (
+            hit["id"] != value_doc.get("id", None)
+            or hit["object_type"] != value_doc.get("object_type", None)
+        )
+    ][0 : settings.OPEN_DISCUSSIONS_SIMILAR_RESOURCES_COUNT]
+
+
```

Comment:
just wondering, we have a similar thing above for getting related posts. in this method we limit the returned 'object_type' to return only learning resources on line 335 - it looks like we don't have a similar limitation on the post code above. do you think we should add one while we're in here? was just thinking that maybe that could end up returning learning resources on the 'related posts' thing, which the UI is currently not set up to handle (and I'm not sure we would want that besides).

Reply:
For some reason that remains a mystery to me, a 'More Like This' query to ES using `like={"_id:" <doc_id>}`, as done in `related_posts`, only returns docs of the same type.  But `like={"doc": <values_doc>}` as used for similar resources, returns multiple types.

Reply:
oh haha, sounds like some obscure elasticsearch quirk